src/FsAutoComplete/FsAutoComplete.Lsp.fs

```diff
@@ -669,86 +676,80 @@ type FsharpLspServer(commands: Commands, lspClient: FSharpLspClient) =
-                                    commands.TryGetLatestTypeCheckResultsForFile(file)
-                                else
-                                    commands.TryGetRecentTypeCheckResultsForFile(file, options) |> async.Return
-
-                        match tyResOpt with
-                        | None ->
-                          logger.info (Log.setMessage "TextDocumentCompletion - no type check results")
-                          return LspResult.internalError "no type check results"
-                        | Some tyRes ->
-                            let! res = commands.Completion tyRes pos lineStr lines file None (config.KeywordsAutocomplete) (config.ExternalAutocomplete)
-                            let res =
-                                match res with
-                                | CoreResponse.Res(decls, keywords) ->
-                                    let items =
-                                        decls
-                                        |> Array.mapi (fun id d ->
-                                            let code =
-                                                if System.Text.RegularExpressions.Regex.IsMatch(d.Name, """^[a-zA-Z][a-zA-Z0-9']+$""") then d.Name
-                                                elif d.NamespaceToOpen.IsSome then d.Name
-                                                else PrettyNaming.QuoteIdentifierIfNeeded d.Name
-                                            let label =
-                                                match d.NamespaceToOpen with
-                                                | Some no -> sprintf "%s (open %s)" d.Name no
-                                                | None -> d.Name
-
-                                            { CompletionItem.Create(d.Name) with
-                                                Kind = glyphToCompletionKind d.Glyph
-                                                InsertText = Some code
-                                                SortText = Some (sprintf "%06d" id)
-                                                FilterText = Some d.Name
-                                                Label = label
-                                            }
-                                        )
-                                    let its = if not keywords then items else Array.append items KeywordList.keywordCompletionItems
-                                    let completionList = { IsIncomplete = false; Items = its}
-                                    success (Some completionList)
-                                | _ ->
-                                  logger.info (Log.setMessage "TextDocumentCompletion - no completion results")
-                                  noCompletion
-                            return res
-                    }
-        return res
-    }
-
-    override __.CompletionItemResolve(ci) = async {
+    override __.TextDocumentCompletion(p: CompletionParams) =
```

Comment:
this is just a refactor to use AsyncResult for clarity (this will probably get rewritten as part of the signature work)

Reply:
I kinda feel bad about `TextDocumentCompletion` in general - probably huge parts of it should be moved to FSAC.Core/Commands.fs layer.

Reply:
I think that with the porting of the stuff phillip did a lot of this will get pushed down a level for sure. Into some domain-agnostic representation, then this layer will just do mapping code.